medcat/utils/regression/regression_checker.py

```diff
@@ -0,0 +1,81 @@
+
+import argparse
+from pathlib import Path
+import logging
+import time
+
+from sys import exit as sys_exit
+from typing import Optional
+
+from medcat.cat import CAT
+from medcat.utils.regression.checking import RegressionChecker, TranslationLayer
+from medcat.utils.regression.results import MultiDescriptor
+
+logger = logging.getLogger(__name__)
+
+
+def main(model_pack_dir: Path, test_suite_file: Path, max_failures=0,
+         total: Optional[int] = None, report: bool = False,
+         phrases: bool = False) -> None:
+    """Check test suite against the specifeid model pack.
+
+    Args:
+        model_pack_dir (Path): The path to the model pack
+        test_suite_file (Path): The path to the test suite YAML
+        total (Optional[int]): The total number of (sub)cases to be tested (for progress bar)
+        report (bool): Whether to use a more comprehensive report
+        phrases (bool): Whether to show per-phrase information in a report
+    """
+    logger.info('Loading RegressionChecker from yaml: %s', test_suite_file)
+    rc = RegressionChecker.from_yaml(str(test_suite_file), report=report)
+    logger.info('Loading model pack from file: %s', model_pack_dir)
+    cat: CAT = CAT.load_model_pack(str(model_pack_dir))
+    logger.info('Checking the current status')
+    st = time.time()
+    # s, f = rc.check_model(cat, TranslationLayer.from_CDB(cat.cdb), total=total)
+    res = rc.check_model(cat, TranslationLayer.from_CDB(cat.cdb), total=total)
+    if report and isinstance(res, MultiDescriptor):
+        logger.info(res.get_report(phrases_separately=phrases))
+        return
+    s, f = res  # tuple
+    logger.info('Checking took %s seconds', time.time() - st)
+    logger.info('\tSuccessful:\t%d\n\tFailed:\t\t%d', s, f)
+    if f > max_failures:
+        logger.warning('Found too many failures (%s)', f)
+        sys_exit(2)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('modelpack', help='The model pack against which to check',
+                        type=Path)
+    parser.add_argument('test_suite', help='YAML formatted file containing the regression test suite'
```

Comment:
probably want to take all .yml files in the configs dir? I'd imagine we'd have multiple test suites that could be run?

Reply:
Can do. But in all honesty, I wasn't entirely sure if the config dir was the correct place for these. Was also contemplating the `examples` folder.

Another possibility would be to ship modelpacks with a certain set of regression tests. And then read the data from there unless explicit other test suite is passed. But that would obviously not work with existing model packs.